awx/main/scheduler/task_manager.py

```diff
@@ -41,6 +41,20 @@
 
 
 logger = logging.getLogger('awx.main.scheduler')
+perf_logger = logging.getLogger('awx.analytics.performance')
+
+
+def perf_logger_timer(func):
+    def inner(*args, **kwargs):
+        start = tz_now()
+        result = func(*args, **kwargs)
+        cls = args[0]
+        perf_logger.debug(
+            f'{{"func": task_manager.{func.__name__}, "duration": {tz_now() - start}, "pending_task_count": {cls.pending_task_count}, "running_task_count": {cls.running_task_count}}}'
```

Comment:
I don't love the handling of this in debug logger statements, because this requires a lot of work for later extraction of the data. I'd rather have us collect the timings in a more complete package, which is bound around the `TaskManager` class, as you're kind of leaning towards here. Ultimately, you want something that can be made into a pie chart at the end of it.

@fosterseth has done some work for efficient collection of stats in subsystem metrics. Since that's how we measure the callback receiver, I think we should use the same tool to measure the task manager timing breakdown.

Either way, it's going to go smoothest if we separate it from the rest of your work here.

Reply:
ok, so move the logger out